node-data/src/events/contract.rs

```diff
@@ -0,0 +1,130 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+//! This module defines the contract event type and related types.
+
+use anyhow::Result;
+use execution_core::ContractId;
+use execution_core::Event;
+use serde::{Deserialize, Deserializer, Serialize, Serializer};
+
+/// Contract event with optional origin (tx hash).
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ContractTxEvent {
+    pub event: ContractEvent,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub origin: Option<[u8; 32]>,
+}
+
+#[derive(Debug, PartialEq, Eq, PartialOrd, Ord, Hash, Clone, Copy)]
+#[repr(C)]
+pub struct WrappedContractId(pub ContractId);
+
+impl Serialize for WrappedContractId {
+    fn serialize<S>(&self, s: S) -> Result<S::Ok, S::Error>
+    where
+        S: Serializer,
+    {
+        let source_hex = hex::encode(self.0.as_bytes());
+        s.serialize_str(&source_hex)
+    }
+}
+
+impl<'de> Deserialize<'de> for WrappedContractId {
+    fn deserialize<D>(deserializer: D) -> Result<WrappedContractId, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let source_hex: String = Deserialize::deserialize(deserializer)?;
+        let source_bytes =
+            hex::decode(source_hex).map_err(serde::de::Error::custom)?;
+        let mut source_array = [0u8; 32];
```

Comment:
where does this number come from? Would it make sense to define a constant for use `mem::size_of`?